eta/core/serial.py

```diff
@@ -197,7 +197,18 @@ def serialize_numpy_array(array):
     # We currently serialize in numpy format. Other alternatives considered
     # were `pickle.dumps(array)` and HDF5
     #
+    header = None
+    if array.dtype == bool:
```

Comment:
What about compressing the serialized bytes using something off-the-shelf?

The concern with introducing custom code here is that it will add extra work to any client who wants to support deserializing these masks for use in their code

Reply:
I'll investigate zlib or something similar as well; this is guaranteed to achieve 8x reduction for arbitrary bit arrays, though (I don't think zlib can achieve that, and introducing zlib on the client side is considerably more complex, while this has already been implemented there)

Reply:
https://github.com/nodeca/pako?

Reply:
Yeah, I think that will work, but we'll need to investigate client-side performance. From initial tests, compressing a single large (1920x1080) mask takes around 150ms - decompressing may be faster.

Reply:
Let's get the system working and then think about implications of CPU vs bandwidth vs interface complexity